File: src/main/java/Day1.java
/*

3   4
4   3
2   5
1   3
3   9
3   3

 */

static final Pattern LINE = Pattern.compile("(\\d+)\\s+(\\d+)\n+");

void main() {
  var input =
          """
          0 0
          """;

  var sortedL = LINE.matcher(input).results()
          .mapToInt(m -> Integer.parseInt(m.group(1)))
          .sorted()
          .toArray();

  var sortedR = LINE.matcher(input).results()
          .mapToInt(m -> Integer.parseInt(m.group(2)))
          .sorted()
          .toArray();

  var distanceSum = IntStream.range(0, sortedL.length)
                  .map(i -> Math.abs(sortedL[i] - sortedR[i]))
                  .sum();

  System.out.println(distanceSum);
}

File: src/main/python/day1.py
import re

input_data = """
3   4
4   3
2   5
1   3
3   9
3   3
"""

pattern = re.compile(r"(\d+)\s+(\d+)")
matches = pattern.findall(input_data)

sorted_l = sorted(int(m[0]) for m in matches)
sorted_r = sorted(int(m[1]) for m in matches)

distance_sum = sum(abs(l - r) for l, r in zip(sorted_l, sorted_r))

print(distance_sum)